woocommerce/woocommerce-ios
# Store creation M2: checkout flow

Part of #8108 
⚠️ Please only review this PR when https://github.com/woocommerce/woocommerce-ios/pull/8205 is ready to merge without critical suggestions ⚠️ 



### Description


We decided to go with a different checkout flow from what we originally planned in p1669122130164769/1668764715.771189-slack-C045CUK1Y3U. After the Yosemite layer changes https://github.com/woocommerce/woocommerce-ios/pull/8205, this PR replaces the current `MockInAppPurchases` with a new `WebPurchasesForWPComPlans` for web checkout when the `storeCreationM2WithInAppPurchasesEnabled` feature flag is disabled. `WebPurchasesForWPComPlans` conforms to `InAppPurchasesForWPComPlansProtocol`, loads the eCommerce monthly plan and creates a cart through `PaymentStore`. After a cart with the eCommerce plan is created, `StoreCreationCoordinator` then shows an authenticated webview to check out with the new `WebCheckoutViewModel`. After a success purchase URL is detected, the coordinator navigates to the in-progress screen to wait for the site to become a Jetpack site and then eventually navigates to the success screen.

### Testing instructions


Please feel free to mention any improvements or issues when testing the flow!

- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> an in-progress modal should be shown briefly, then dismissed. the store name form is then presented
- Enter some store name and tap `Continue` when ready --> the store name should be used for the initial domain query in the next screen
- Search for another domain if you like, and select one domain to `Continue` --> a free Simple site should be created, and a store summary screen is shown with the store name and site slug
- Tap `Continue to payment` --> the eCommerce plan should be shown in the currency of the WPCOM account
- Tap `Create Store for */month` --> an in-progress screen should be shown about creating the store. after the site becomes available as a Jetpack site in the `/me/sites` response (about 30s-1m), the success screen should be shown
- Tap `Manage My Store` to continue to the new site in the app --> the new site should be an Atomic site with WC activated, where you can add a product successfully

### Screenshots


Example screencast in p1669190357377079/1668764715.771189-slack-C045CUK1Y3U

---
